overlord/fdestate/backend/reseal.go

```diff
@@ -445,42 +480,16 @@ func updateRunProtectionProfile(
 
 	logger.Debugf("PCR profile length: %v", len(pcrProfile))
 
-	var models []secboot.ModelForSealing
-	for _, m := range modelParams {
-		models = append(models, m.Model)
-	}
-
-	var modelsRunOnly []secboot.ModelForSealing
-	for _, m := range modelParamsRunOnly {
-		modelsRunOnly = append(modelsRunOnly, m.Model)
-	}
-
-	runParams := &SealingParameters{
-		BootModes:     []string{"run", "recover"},
-		Models:        models,
-		TpmPCRProfile: pcrProfile,
-	}
-
-	// TODO:FDEM: use constants for "run+recover" and "all"
-	if err := manager.Update("run+recover", "all", runParams); err != nil {
-		return err
-	}
-
-	runOnlyParams := &SealingParameters{
-		BootModes:     []string{"run"},
-		Models:        modelsRunOnly,
-		TpmPCRProfile: pcrProfileRunOnly,
-	}
-	// TODO:FDEM: use constants for "run+recover" and "all"
-	if err := manager.Update("run", "all", runOnlyParams); err != nil {
-		return err
-	}
+	parameters.get("run+recover", "all").TpmPCRProfile = pcrProfile
```

Comment:
I think, as this is the main thing we have it for, updateParameters should have an updatePcrProfile method doing this